modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/raft/RocksDbClusterStateStorage.java

```diff
@@ -68,33 +71,46 @@ public class RocksDbClusterStateStorage implements ClusterStateStorage {
 
     private final Object snapshotRestoreLock = new Object();
 
-    public RocksDbClusterStateStorage(Path dbPath) {
+    /**
+     * Creates a new instance.
+     *
+     * @param dbPath Path to the database.
+     * @param nodeName Ignite node name.
+     */
+    public RocksDbClusterStateStorage(Path dbPath, String nodeName) {
         this.dbPath = dbPath;
+        this.snapshotExecutor = Executors.newSingleThreadExecutor(
+                NamedThreadFactory.create(nodeName, "cluster-state-snapshot-executor", LOG)
+        );
     }
 
     @Override
     public CompletableFuture<Void> start() {
-        options = new Options().setCreateIfMissing(true);
+        init();
+
+        return nullCompletedFuture();
+    }
 
+    private void init() {
         try {
-            db = RocksDB.open(options, dbPath.toString());
+            RocksDB db = RocksDB.open(options, dbPath.toString());
 
             ColumnFamily defaultCf = ColumnFamily.wrap(db, db.getDefaultColumnFamily());
 
             snapshotManager = new RocksSnapshotManager(db, List.of(fullRange(defaultCf)), snapshotExecutor);
+
+            this.db = db;
         } catch (RocksDBException e) {
-            throw new IgniteInternalException("Failed to start the storage", e);
+            throw new StorageException("Failed to start the storage", e);
```

Comment:
Why StorageException? IgniteInternalException looks better

Reply:
It's deprecated